### ISSUE TITLE
Cannot render if joined with other geometries

### ISSUE THREAD

**Issue by hertz98** [xypwn/scadqr#8]
Hello

I like your project, in fact I have used it on [this project](https://www.thingiverse.com/thing:6694276) and its generator works fine

I'm not sure if it is a OpenSCAD issue or with your library, when **rendering** it's hard (sometimes impossible) to get the qr code join to another geometry without getting `ERROR: The given mesh is not closed! Unable to convert to CGAL_Nef_Polyhedron. `, it's fine in the preview

I'm using OpenSCAD version 2021.01-x86-64 on Windows 10 x64

In the next example the qr code is not rendered

```OpenSCAD
include <qr.scad>

translate([0,0,-5/2])
    cube([60,60,5], center = true);

color("black")
    qr( qr_wifi(ssid = "WiFi_SSID", 
                psk = "WiFi_Password", 
                auth = "WPA", 
                hidden = false),
        error_correction = "H",
        height = 50,
        width = 50,
        thickness = 1,
        center = true);
```

A workaround is to project, and then extrude again (but it does not work extruding the 2D geometry directly by setting thickness=0)

```OpenSCAD
include <qr.scad>

translate([0,0,-5/2])
    cube([60,60,5], center = true);

color("black")
    linear_extrude(1)
        projection()
            qr( qr_wifi(ssid = "WiFi_SSID", 
                        psk = "WiFi_Password", 
                        auth = "WPA", 
                        hidden = false),
                error_correction = "H",
                height = 50,
                width = 50,
                thickness = 1,
                center = true);
```

but even in this way there are some cases where still does not work, at example I could set the error correction to "M" or "L"

```OpenSCAD
include <qr.scad>

translate([0,0,-5/2])
    cube([60,60,5], center = true);

color("black")
    linear_extrude(1)
        projection()
            qr( qr_wifi(ssid = "WiFi_SSID", 
                        psk = "WiFi_Password", 
                        auth = "WPA", 
                        hidden = false),
                error_correction = "M",
                height = 50,
                width = 50,
                thickness = 1,
                center = true);
```

Exporting the mesh, and then importing it, cause the same behavior, the slicer complains about open edges on the mesh

**Comment by xypwn** [comment]
Hi, thanks for reporting this!

I can reproduce the issue on OpenSCAD 2021.01. It's related to https://github.com/openscad/openscad/issues/5046 , where the renderer gets confused because the edges of diagonally adjacent QR code modules are too close together.
This has been fixed as of 2024-05-01 (I confirmed that this problem does not occur in 2024.08.05).

I've increased the module scaling summand by factor 10, which seems to alleviate this problem, including in older versions. Let me know if the new version of this library works for you too.
